src/core/client.ts

```diff
@@ -97,11 +100,14 @@ export default class Client {
       this.config.__plugins.forEach((plugin) => plugin.load(this))
     }
 
-    this.__store = opts.store || new DefaultStore({ context: {}, breadcrumbs: [] })
```

Comment:
Is there any reason you moved this from here to the constructor? At first, I was conflicted as well, but then I realized that `Honeybadger.configure` is a required method call for it to work (you need to at least call it to set the `apiKey`). And since we have other initializations happening here, I decided to include the store initialization.

We can leave it as you want, it doesn't make much difference, I just want to know how you thought about this.

Finally, if we are keeping this version, we should remove the `store` property from the `Config` interface (`types.ts`).

Reply:
For one, the `.configure` method is typically for things the end user should set, and for now, the store is not meant to be set by them. It's too critical and tied to implementation details. And they don't need to set it either, as `hb.run` will set it as needed.

Also, the store is too important to not be set, IMO. It's better to construct the object straightaway with as many key details set as possible, avoiding any invalid states.

Finally, in our tests, we don't always call `.configure`, and having to add that to get things working was a pain.

Reply:
> For one, the .configure method is typically for things the end user should set, and for now, the store is not meant to be set by them. It's too critical and tied to implementation details. And they don't need to set it either, as hb.run will set it as needed.

Good point.

> Also, the store is too important to not be set, IMO. It's better to construct the object straightaway with as many key details set as possible, avoiding any invalid states.

I agree, but the client should not work without a call to configure. Perhaps we should enforce this 🤔 (i.e. throw an error if `configure` is not called)? Or go the other way make the `configure` method completely optional (i.e. read the `apiKey` from .env)?

> Finally, in our tests, we don't always call .configure, and having to add that to get things working was a pain.

Hehe, I faced the same scenario. I went ahead and added `configure` because otherwise we were not initializing the client correctly.

Anyway, I'm convinced 😄 and I think @joshuap is too, because he suggested adding the store initialization in the constructor too. I just didn't go for it to keep initializations in one place.

Reply:
Since we're an error-reporting library, we can't afford to throw an error😅

At best we can write a log if there's no API key when we try to report, but I think we maybe already do that.

I don't think we need to change anything here at the moment. I think expecting the user to call `configure` is fair. It's always in the usage instructions, right next to the `require()` part.